### DISCUSSION TITLE
Why does dracut remove wireless modules?

### DISCUSSION THREAD

**Discussion by crempel-redhat** [dracutdevs/dracut#2514] [General]
I'd like to be able to bring up wifi in the initramfs to work with NBDE but dracut removes wireless support:

https://github.com/dracutdevs/dracut/blob/056/modules.d/90kernel-network-modules/module-setup.sh#L15

First I'm wanting to understand why it is removed and second, I'd like to learn what would be involved to DIY this, unless it is not possible for some reason.

Thanks!

**Comment by aafeijoo-suse** [comment]
> First I'm wanting to understand why it is removed

This is really old stuff (https://github.com/dracutdevs/dracut/commit/9dcf9cef387f7582eb8e8d6632d768411ead3617). Although it has no explanation, I assume the reason is that it adds drivers that are not really necessary for a common use case.

> and second, I'd like to learn what would be involved to DIY this, unless it is not possible for some reason.

As the referenced commit says, you can create a custom dracut module (i.e. `90kernel-network-wireless-modules` or so), based on the `90kernel-network-modules` module, filling the `installkernel()` function with the drivers you want to add.

**Comment by krpage** [comment]
I can report limited achievements on Fedora (so using the network-manager module in dracut): I needed to include the wireless kernel modules (iwlwifi, iwlmvm in my case) through a dracut config, I also needed to manually include files from the firmware package (e.g. iwlwifi-so-a0-gf-a0.pnvm.xz) which weren't being picked up automatically by dracut, and libnm-device-plugin-wifi.so from the NetworkManager-wifi package. At this point nmcli on the dracut emergency shell could see and recognise the wireless device, however it doesn't activate it despite my also including the relevant connection file from /etc/NetworkManager/system-connections/

I see SuSE has (or had?) a dracut-wireless package, suggesting this needs more than the above to work.

**Comment by LaszloGombos** [comment]
CC @nadvornik (person involved with dracut-wireless package) . Thanks !
